server/service/scenarios/db.js

```diff
@@ -15,17 +11,39 @@ exports.getScenario = async function getScenario(scenarioId) {
             where scenario_id = ${scenarioId};`
     );
 
+    return scenarioCategoriesResults.rows.map(r => r.name);
+}
+
+exports.getScenario = async function getScenario(scenarioId) {
+    const results = await query(
+        sql` SELECT * FROM scenario WHERE id = ${scenarioId};`
+    );
+
+    const categories = await getScenarioCategories(scenarioId);
+
     return {
         ...results.rows[0],
-        categories: scenarioCategoriesResults.rows.map(r => r.name)
+        categories
     };
 };
 
 exports.getAllScenarios = async function getAllScenarios() {
     const results = await query(
         sql`SELECT * FROM scenario ORDER BY created_at DESC;`
     );
-    return results.rows;
+
+    async function getResultsWithCategories() {
```

Comment:
Out of curiosity, why is this but of logic in its own function? Not saying it's wrong, just to understand.

Reply:
I put it in its own function because the mechanics were very specific to a list of scenarios. I wanted the exported functions to deal with a single scenario only.

Reply:
I think what @ktbee is getting at is that you don't need to wrap this in a function at all. I believe the following has the same output: 

```js
exports.getAllScenarios = async function getAllScenarios() {
    const results = await query(
        sql`SELECT * FROM scenario ORDER BY created_at DESC;`
    );

    let resultsWithCategories = [];
    for (let r of results.rows) {
        let categories = await getScenarioCategories(r.id);
        resultsWithCategories.push({ ...r, categories });
    }

    return resultsWithCategories;
};
```

Reply:
Got it! Will fix

Reply:
👍 thanks for clarifying, thought I was missing something